### PR TITLE
Add example for kubeflow spot instance cluster config

### PR DESCRIPTION
**Pull request by arjun921**
Helps users build a highly cost efficient Kubeflow cluster for ML/DL training at 50-80% less cost

### Description

Added cost efficient Kubeflow cluster spec with 
- Spot GPU instances
- Scale down to zero

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [ ] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

**Comment by arjun921** [comment]
kind/improvement

**Comment by arjun921** [comment]
Label: kind/improvement

**Comment by arjun921** [comment]
Hi @martina-if ,

That's a really good catch! 

I'll make those changes and push soon. 

I have one doubt though, would it be helpful if I included a detailed blog post as a comment in the file?

Something like line 3 on the source: https://github.com/arjun921/aws-spot-instances-kubeflow/blob/master/envs/staging/cluster-spec.yml

**Comment by martina-if** [comment]
Yeah, I think the link could be useful for users :+1:

File: examples/23-kubeflow-spot-instance.yaml [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+      # maxPrice: 4.4
+      instanceTypes: ["p3.8xlarge"]
+      onDemandBaseCapacity: 0
+      onDemandPercentageAboveBaseCapacity: 0
+      spotAllocationStrategy: capacity-optimized
+    labels:
+      lifecycle: Ec2Spot
+      aws.amazon.com/spot: "true"
+      gpu-count: "4"
+    # Stick to one AZ for all GPU nodes. In case of termination, this will prevent volumes from being unavailable if the new instance got spun up in another AZ.
```
Comment (martina-if) [inline examples/23-kubeflow-spot-instance.yaml:162]: nit: fix this line too please :)

Comment (arjun921) [reply examples/23-kubeflow-spot-instance.yaml:162]: Oops, missed that one😶
Fixing right away!

Comment (arjun921) [reply examples/23-kubeflow-spot-instance.yaml]: @martina-if Done

**Comment by arjun921** [comment]
@martina-if All requested changes done!
Please let me know if there's anything that I might have missed out on.

**Review by martina-if — APPROVED** [review]

**Comment by arjun921** [comment]
@martina-if I've updated test cases, 
it's hopefully ready for a merge 😬

**Pull request merged**